chain/gen/genesis/miners.go

```diff
@@ -158,22 +157,23 @@ func SetupStorageMiners(ctx context.Context, cs *store.ChainStore, sroot cid.Cid
 				return cid.Undef, xerrors.Errorf("flushing vm: %w", err)
 			}
 
-			mact, err := vm.StateTree().GetActor(minerInfos[i].maddr)
-			if err != nil {
-				return cid.Undef, xerrors.Errorf("getting newly created miner actor: %w", err)
-			}
+			//mact, err := vm.StateTree().GetActor(minerInfos[i].maddr)
+			//if err != nil {
+			//return cid.Undef, xerrors.Errorf("getting newly created miner actor: %w", err)
+			//}
 
-			mst, err := miner.Load(adt.WrapStore(ctx, cst), mact)
-			if err != nil {
-				return cid.Undef, xerrors.Errorf("getting newly created miner state: %w", err)
-			}
+			//mst, err := miner.Load(adt.WrapStore(ctx, cst), mact)
+			//if err != nil {
+			//return cid.Undef, xerrors.Errorf("getting newly created miner state: %w", err)
+			//}
 
-			pps, err := mst.GetProvingPeriodStart()
-			if err != nil {
-				return cid.Undef, xerrors.Errorf("getting newly created miner proving period start: %w", err)
-			}
+			//pps, err := mst.GetProvingPeriodStart()
+			//if err != nil {
+			//return cid.Undef, xerrors.Errorf("getting newly created miner proving period start: %w", err)
+			//}
 
-			minerInfos[i].presealExp = (maxPeriods-1)*miner0.WPoStProvingPeriod + pps - 1
+			//minerInfos[i].presealExp = (maxPeriods-1)*miner0.WPoStProvingPeriod + pps - 1
```

Comment:
@arajasek this comes up to `129537` which is ~45 days, so when we submit initial deals for presealing (`chain/gen/genesis/miners.go:193`), the VM throws an error.

Is this value supposed to be equal to 45 days?

Reply:
@arajasek can you explain what is the idea here with `(maxPeriods-1)*miner0.WPoStProvingPeriod + pps - 1` ? We seem to be using this value for the initial deal durations when building preseal sectors, and this is not passing the 180 days validation.

Reply:
@nonsense I actually don't know why we do `(maxPeriods-1)*miner0.WPoStProvingPeriod + pps - 1`, looking through history it seems like it might be legacy (we've "always" done it).

The value definitely shouldn't be ~45 days, though, it should be ~540 days. We need to figure out why that's happening -- it might be because of some funkiness with block times. Does lotus soup use a custom block time?

Reply:
Will try to repro on devnet.

Reply:
Hmm, confirmed it gets 1550097 on a devnet, so I think this is a lotus-soup config thing.

Reply:
@arajasek thanks, that makes sense now - we do reduce the window in the test plan, so that we can test proving failures - it happens at https://github.com/filecoin-project/lotus/blob/master/testplans/lotus-soup/init.go#L45

I think we will want to keep the flexibility to change the proving period, but it looks like it is not correlated to the minimum deal duration...